src/axom/quest/examples/quest_distributed_distance_query_example.cpp

```diff
@@ -0,0 +1,881 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file quest_distributed_distance_query_example.cpp
+ * \brief Driver for a distributed distance query
+ */
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+#include "axom/sidre.hpp"
+#include "axom/quest.hpp"
+#include "axom/slam.hpp"
+
+#include "conduit_blueprint.hpp"
+#include "conduit_blueprint_mpi.hpp"
+
+#include "axom/quest/DistributedClosestPoint.hpp"
+
+#include "axom/fmt.hpp"
+#include "axom/CLI11.hpp"
+
+#ifndef AXOM_USE_MFEM
+  #error This example requires Axom to be configured with MFEM and the AXOM_ENABLE_MFEM_SIDRE_DATACOLLECTION option
+#endif
+#include "mfem.hpp"
+
+#ifndef AXOM_USE_MPI
+  #error This example requires Axom to be configured with MPI
+#endif
+#include "mpi.h"
+
+// RAJA
+#ifdef AXOM_USE_RAJA
+  #include "RAJA/RAJA.hpp"
+#endif
+
+// C/C++ includes
+#include <string>
+#include <map>
+#include <cmath>
+
+namespace quest = axom::quest;
+namespace slic = axom::slic;
+namespace sidre = axom::sidre;
+namespace slam = axom::slam;
+namespace spin = axom::spin;
+namespace primal = axom::primal;
+namespace mint = axom::mint;
+namespace numerics = axom::numerics;
+
+using RuntimePolicy = axom::quest::DistributedClosestPoint::RuntimePolicy;
+
+/// Struct to parse and store the input parameters
+struct Input
+{
+public:
+  std::string meshFile;
+
+  double circleRadius {1.0};
+  int circlePoints {100};
+  RuntimePolicy policy {RuntimePolicy::seq};
+
+private:
+  bool m_verboseOutput {false};
+
+  // clang-format off
+  const std::map<std::string, RuntimePolicy> s_validPolicies{
+    #if defined(AXOM_USE_RAJA) && defined(AXOM_USE_UMPIRE)
```

Comment:
Why is RAJA and Umpire guarding seq? Shouldn't that be there regardless?  What happens when all 4 of these are off? I don't see this example guarded by any of these in the CMake but I could have missed it.

Reply:
Thanks -- I fixed the guards -- this query no longer requires RAJA or Umpire for the sequential policy